apps/flair/lib/flair/durations/consumer.ex

```diff
@@ -0,0 +1,73 @@
+defmodule Flair.Durations.Consumer do
+  @moduledoc """
+  Consumer is called at the end of a flow; it converts events and then passes them to presto client to be persisted.
+  """
+
+  use GenStage
+
+  @table_writer Application.get_env(:flair, :table_writer)
+  @table_name Application.get_env(:flair, :table_name_timing, "operational_stats")
+
+  @table_schema [
+    %{name: "dataset_id", type: "string"},
+    %{name: "app", type: "string"},
+    %{name: "label", type: "string"},
+    %{name: "timestamp", type: "long"},
+    %{
+      name: "stats",
+      type: "map",
+      subSchema: [
+        %{name: "count", type: "long"},
+        %{name: "min", type: "double"},
+        %{name: "max", type: "double"},
+        %{name: "std", type: "double"},
+        %{name: "average", type: "double"}
+      ]
+    }
+  ]
+
+  def start_link(name, args \\ nil) do
+    GenStage.start_link(__MODULE__, args, name: name)
+  end
+
+  def init(_args) do
+    {:consumer, :any}
+  end
+
+  def handle_events(events, _from, state) do
+    events
+    |> convert_events()
+    |> @table_writer.write(table: @table_name, schema: @table_schema)
+
+    {:noreply, [], state}
+  end
+
+  def bootstrap() do
+    case Application.get_env(:flair, :table_name_timing) do
+      nil -> :ok
+      _ -> @table_writer.init(table: @table_name, schema: @table_schema)
+    end
+  end
+
+  defp convert_events(events) do
+    events
+    |> Enum.map(fn {dataset_id, stats_map} ->
```

Comment:
is there any way to use a reduce on this to avoid the nested maps? is there any chance this operation will need to be faster than a map over a map operation can keep up with?

Reply:
Probably. I renamed this module and moved the file, but I didn't write this code. I didn't want to touch it as part of this PR.